gtsam/navigation/AttitudeFactor.md

```diff
@@ -0,0 +1,142 @@
+# AttitudeFactor in GTSAM
+
+[Cautionary note: this was generated from the source using ChatGPT but edited by Frank]
+
+## Introduction
+
+The `AttitudeFactor` in GTSAM is a factor that constrains the orientation (attitude) of a robot or sensor platform based on directional measurements. This is particularly useful in GPS-denied navigation contexts where orientation must be estimated from inertial sensors like accelerometers or magnetometers.
+
+This document explains the mathematical foundation of the `AttitudeFactor` and guides users on how to use this factor effectively in GTSAM.
+
+## Mathematical Foundation
+
+### Concept
+
+The `AttitudeFactor` constrains the rotation $\mathbf{R}_{nb}$ (from body frame $b$ to navigation frame $n$) such that a known reference direction in the body frame aligns with a measured direction in the navigation frame when rotated. The factor enforces that:
+
+$$
+\mathbf{R}_{nb} \cdot \mathbf{b}_{\text{Ref}} \approx \mathbf{nZ}
+$$
+
+where:
+
+- $\mathbf{R}_{nb}$ is the rotation matrix representing the orientation from body to navigation frame.
+- $\mathbf{b}_{\text{Ref}}$ is a known reference direction in the body frame (e.g., the accelerometer's sensitive axis).
+- $\mathbf{nZ}$ is the measured direction in the navigation frame (e.g., the gravity vector measured by an IMU).
+
+### Error Function
+
+The error function computes the angular difference between the rotated reference direction and the measured direction:
+
+$$
+\mathbf{e} = \text{error}(\mathbf{nZ}, \mathbf{R}_{nb} \cdot \mathbf{b}_{\text{Ref}})
+$$
+
+This error is minimal (zero) when the rotated body reference direction aligns perfectly with the measured navigation direction.
+
+The error is computed internally using the `attitudeError` function:
+
+```cpp
+Vector AttitudeFactor::attitudeError(const Rot3& nRb) const {
+  Unit3 nRef = nRb * bRef_;
+  return nZ_.error(nRef);
+}
+```
+
+#### Explanation:
+- The function computes the rotated reference direction `nRef` and then the error between `nRef` and the measured direction `nZ`.
+- `nZ_.error(nRef)` is a 2D vector-valued error between two directions, defined in [Unit3.h](../geometry/Unit3.h).
+
+### Jacobians
+
+For optimization, the $2 \times 3$ Jacobian of the error function with respect to the rotation parameters is required. The Jacobian is computed using chain rule differentiation, involving the derivative of the rotated vector with respect to the rotation parameters and the derivative of the error with respect to the rotated vector.
```

Comment:
Maybe a mention here of the singularity we found w.r.t. the Jacobian, which also vanishes 180 degrees away from the true direction (because it is calculated in the tangent space?)

Reply:
Done